manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -95,9 +101,7 @@ public Bridge createDependencies(Work work, Bridge bridge) {
                 // This ensures Users cannot update the Bridge Error Handler until it is available.
                 bridge.setDependencyStatus(ManagedResourceStatus.READY);
             } else if (errorHandler.getStatus() == ManagedResourceStatus.FAILED) {
-                bridge.setStatus(ManagedResourceStatus.FAILED);
-                bridge.setDependencyStatus(ManagedResourceStatus.FAILED);
-                propagateProcessorError(bridge);
+                return persistAndPropagateErrorHandlerFailure(bridge, errorHandler);
```

Comment:
nitpick: I'd do two different methods, one for persisting and the other to propagate the error as they seem unrelated

Reply:
I do the same thing (persist and propagate) in two different places. 

The single method was added to remove duplicate code.